### PR TITLE
[BUG] adjust CFRadial's _is_valid

### LINKED ISSUE #3987: CF radial frontend is causing SIGSEGV for certain files
### Bug report

**Bug summary**

CFRadial uses `xarray.open_dataset` in `_is_valid` and due to https://github.com/pydata/xarray/issues/6728 it causes SIGSEGV in `yt.load` for certain files (e.g. `GasSloshing/sloshing_nomag2_hdf5_plt_cnt_0150`)

**Code for reproduction**

```python

from yt.frontends.cf_radial.data_structures import CFRadialDataset

CFRadialDataset._is_valid("GasSloshing/sloshing_nomag2_hdf5_plt_cnt_0150")

```

**Actual outcome**

```
$ python3 -Wd -X faulthandler bug.py 
Fatal Python error: Segmentation fault

Current thread 0x00007fe5cdb3c740 (most recent call first):
  File "/home/xarth/codes/xarthisius/xarray/xarray/backends/netCDF4_.py", line 106 in _getitem
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/indexing.py", line 816 in explicit_indexing_adapter
  File "/home/xarth/codes/xarthisius/xarray/xarray/backends/netCDF4_.py", line 93 in __getitem__
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/indexing.py", line 527 in __array__
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/variable.py", line 250 in _as_array_or_item
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/variable.py", line 510 in values
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/variable.py", line 337 in data
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/common.py", line 1866 in contains_cftime_datetimes
  File "/home/xarth/codes/xarthisius/xarray/xarray/core/common.py", line 1875 in _contains_datetime_like_objects
  File "/home/xarth/codes/xarthisius/xarray/xarray/conventions.py", line 345 in decode_cf_variable
  File "/home/xarth/codes/xarthisius/xarray/xarray/conventions.py", line 521 in decode_cf_variables
  File "/home/xarth/codes/xarthisius/xarray/xarray/backends/store.py", line 27 in open_dataset
  File "/home/xarth/codes/xarthisius/xarray/xarray/backends/netCDF4_.py", line 567 in open_dataset
  File "/home/xarth/codes/xarthisius/xarray/xarray/backends/api.py", line 495 in open_dataset
  File "/home/xarth/codes/xarthisius/yt/bug.py", line 10 in <module>
Segmentation fault (core dumped)
```

**Expected outcome**

`_is_valid` not to die.

**Notes**
Happens with `xarray > 2022.3.1`

### PR DESCRIPTION
**Pull request by chrishavlin**
This switches CFRadial's `_is_valid` method to use yt's  `NetCDF4FileHandler` instead of xarray following the nc4_cm1 frontend. xarray is still used for opening and loading data after the dataset is identified as a CFRadial dataset. 

Closes #3987

### CODE REVIEW

**Comment by chrishavlin** [comment]
Extra note -- since the underlying issue this PR is addressing can be reproduced when opening with just the `netCDF4` library ([link](https://github.com/pydata/xarray/issues/6728#issuecomment-1167344851)), I just installed the latest netcdf4 from pip (v1.6.0) and confirmed the following still works on this branch:

```python
from yt.frontends.cf_radial.data_structures import CFRadialDataset
CFRadialDataset._is_valid("GasSloshing/sloshing_nomag2_hdf5_plt_cnt_0150")
# prints False
```

I think because `_is_valid` here is checking simple attributes we should be safe from the underlying issue. 

@Xarthisius any more thorough tests I should try?

**Review by neutrinoceros — COMMENTED** [review]
LGTM, but as I'm not able to reproduce the issue (maybe it doesn't happen on OsX ?), I'll defer to others for approvals

File: yt/frontends/cf_radial/data_structures.py
Comment (chrishavlin) [reply yt/frontends/cf_radial/data_structures.py:310]: well the previous `_is_valid` would sum if both existed so adding a break would change the behavior from what `_is_valid` was doing here... but that's probably OK... @zssherman -- is there any case in which a cfradial file might have both a "Conventions" and "conventions" attribute?

Comment (zssherman) [reply yt/frontends/cf_radial/data_structures.py:310]: However, it shouldn't have both

Comment (chrishavlin) [reply yt/frontends/cf_radial/data_structures.py:310]: Oops, looks like we **do** need to always check both. Or only check `conventions`. I opened up one of the gridded files that gets saved off and it actually has both:

```
>>> import netCDF4 as nc
>>> ds = nc.Dataset('swx_20120520_0641_yt_grid.nc', mode="r")
>>> ds.Conventions
'PyART_GRID-1.1'
>>> ds.conventions
'CF/Radial'
```

maybe it'd be fine just checking `conventions`. But think I'll just switch it back to checking both so we can get the bug that this PR is addressing fixed and we can revisit this question later.

**Comment by Xarthisius** [comment]
> I checked for obvious offenders, and I report that neither `netcdf4` or `xarray` shipped a release today (nor did they originate this constraint). I'm puzzled.

https://github.com/ARM-DOE/pyart/blob/a7170898616543c28b758c951c45f065156a80e5/requirements.txt#L6

from arm-pyart==1.12.5

**Comment by zssherman** [comment]
@Xarthisius Hmm yeah we did a pin of cftime, @mgrover1 what was the original reason for the pin? Can always do a reverse and do a micro release to fix cftime pinning, or find a work around.

**Comment by neutrinoceros** [comment]
I also note pyart is pinning numpy. This is most likely going to be a problem for us too, we don't want to constraint a crucial dependency to satisfy an optional one.

**Comment by mgrover1** [comment]
We were running into an issue with cftime after they added some new Cython code to speed things up - the recent NumPy release was causing some errors, but it looks like this is being fixed on the cftime side [PR 281](https://github.com/Unidata/cftime/pull/281).. once that get's merged + released, we can unpin <1.23.0 for NumPy in Py-ART

**Comment by zssherman** [comment]
@chrishavlin Sounds good, ill be pushing out a new pyart release here shortly as well.

**Comment by neutrinoceros** [comment]
I think @Xarthisius should validate that this fixes the issue

**Review by Xarthisius — APPROVED** [review]

**Pull request merged**
